src/tests/grid/test_stretchedRegular.cc

```diff
@@ -0,0 +1,228 @@
+                                         -6.311050724137887258,  -5.659902448275774489, -5.00875417241366172,
+                                         -4.357605896551548952,  -3.706457620689436183, -3.055309344827323415,
+                                         -2.404161068965210646,  -1.753012793103097877, -1.101864517240985109,
+                                         -0.4507162413788723399, 0.2004320344832404288, 0.8515803103453531975,
+                                         1.502728586207465966,   2.153876862069578735,  2.805025137931691503,
+                                         3.456173413793804272,   4.107321689655917041,  4.758469965518029809,
+                                         5.409618241380142578,   6.060766517242255347,  6.711914793104368115,
+                                         7.363063068966480884,   8.014211344828593653,  8.665359620690706421,
+                                         9.31650789655281919,    9.967656172414931959,  10.61880444827704473,
+                                         11.269952724139157,     11.92110100000127};
+
+const int nx               = 34;
+const int ny               = 32;
+const double xrange[2]     = {348.13120344827576, 369.6190965517242};
+const double yrange[2]     = {-8.264495551724226, 11.92110100000127};
+const double xrange_reg[2] = {351.386944827586319, 366.363355172413776};
+const double yrange_reg[2] = {-5.008754172413662, 8.665359620690706};
+
+const double rim_width = 4.;
+const double delta_low = 1.;
+const double delta_hi  = 0.6511482758621128;
+
+auto make_var_ratio_projection = [](double var_ratio) {
+    Config conf;
+    conf.set("type", "stretch");
+    conf.set("var_ratio", var_ratio);
+    conf.set("delta_low", delta_low);
+    conf.set("delta_hi", delta_hi);
+    conf.set("x_reg_start", xrange_reg[0]);
+    conf.set("x_reg_end", xrange_reg[1]);
+    conf.set("y_reg_start", yrange_reg[0]);
+    conf.set("y_reg_end", yrange_reg[1]);
+    conf.set("startx", xrange[0]);
+    conf.set("endx", xrange[1]);
+    conf.set("starty", yrange[0]);
+    conf.set("endy", yrange[1]);
+    conf.set("north_pole", {0.0, 90.0});
+    conf.set("rim_widthx", rim_width);
+    conf.set("rim_widthy", rim_width);
+    return atlas::Projection(conf);
+};
+
+
+};  // namespace
+
+
+namespace atlas {
+namespace test {
+
+CASE("print delta") {
+    const std::vector<double>& v = lon_LAM_str;
+    std::vector<double> delta(v.size() - 1);
+    for (size_t i = 0; i < delta.size(); ++i) {
+        delta[i] = v[i + 1] - v[i];
+    }
+    Log::info() << "delta = " << delta << std::endl;
+    // Outputs:
+    // delta = [1,1,0.883845,0.798259,0.720961,0.651148,0.651148,0.651148,0.651148,
+    //          0.651148,0.651148,0.651148,0.651148,0.651148,0.651148,0.651148,0.651148,
+    //          0.651148,0.651148,0.651148,0.651148,0.651148,0.651148,0.651148,0.651148,
+    //          0.651148,0.651148,0.651148,0.720961,0.798259,0.883845,1,1]
+}
```

Comment:
Hi @MO-marcomilan I added this "CASE" in the test must to understand a bit the progression.
As I understand from the configuration, the "var_ratio = 1.13" which means that in the transition zone the delta variables should progress with a factor 1.13, but when looking at the values it is not the case.
Could you elaborate?

Reply:
Hi, thank you for the comment.

A comment was missing in StretchRegular.cc, it is a part of the original algorithm (from Terry Davies).
/** compute ratio,
           *  change stretching factor so that high and low grids
           *   retain original sizes
           */
          double var_ints_f = (n_int - n_rim_- ints_high)/2.;
          double logr = log(var_ratio_);
          double log_ratio = (var_ints_f - 0.5) * logr;
          double new_ratio = exp(log_ratio / var_ints);

In this case the actual ratio (new_ratio) is around 1.10722.

Reply:
Thank you very much for that addition, and apologies for my ignorance, but could you perhaps also in words explain what following variables indicate?
- var_ints
- var_ints_f
- n_int
- n_rim
- ints_high
Perhaps a sketch or drawing would help?

Reply:
![image](https://user-images.githubusercontent.com/71659910/145207779-77022ff1-4f5a-4559-a958-6256e09f2d92.png)

There are some comments near the variable definitions, here a simple sketch (I hope that you can see it)

Reply:
Thank you that is very helpful